includes/class-amp-theme-support.php

```diff
@@ -1243,4 +1252,112 @@ public static function enqueue_assets() {
+			return $html;
+		};
+
+		if ( ! has_header_video() ) {
+			return self::output_header_image( $atts );
+		}
+
+		return self::output_header_video( $atts );
+	}
+
+	/**
+	 * Replace the header image markup with a header video.
+	 *
+	 * This is JS-driven in Core themes like Twenty Sixteen and Twenty Seventeen.
+	 * So in order for the header video to display,
+	 * this replaces the markup of the header image.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/WordPress/wordpress-develop/blob/d002fde80e5e3a083e5f950313163f566561517f/src/wp-includes/js/wp-custom-header.js#L54
+	 * @param array $atts The header tag attributes array.
+	 * @return string $html Filtered markup.
+	 */
+	public static function output_header_video( $atts ) {
+		// Remove the script for video.
+		wp_deregister_script( 'wp-custom-header' );
+		$video_settings = get_header_video_settings();
+
+		$parsed_url       = wp_parse_url( $video_settings['videoUrl'] );
+		$query            = isset( $parsed_url['query'] ) ? wp_parse_args( $parsed_url['query'] ) : null;
+		$video_attributes = array(
+			'media'    => '(min-width: ' . $video_settings['minWidth'] . 'px)',
+			'width'    => $video_settings['width'],
+			'height'   => $video_settings['height'],
+			'layout'   => 'fill',
+			'autoplay' => '',
+			'id'       => 'wp-custom-header-video',
+		);
+
+		$atts['placeholder'] = '';
+		$image_placeholder   = self::output_header_image( $atts );
+
+		// If the video URL is for YouTube, return an <amp-youtube> element.
+		if ( isset( $parsed_url['host'], $query['v'] ) && ( false !== strpos( $parsed_url['host'], 'youtube' ) ) ) {
+			$video_header = AMP_HTML_Utils::build_tag(
+				'amp-youtube',
+				array_merge(
+					$video_attributes,
+					array(
+						'data-videoid'        => $query['v'],
+						'data-param-rel'      => '0', // Don't show related videos.
+						'data-param-showinfo' => '0', // Don't show video title at the top.
+						'data-param-controls' => '0', // Don't show video controls.
+					)
+				)
+			);
+		} else {
+			$video_header = AMP_HTML_Utils::build_tag(
+				'amp-video',
+				array_merge(
+					$video_attributes,
+					array(
+						'src' => $video_settings['videoUrl'],
+					)
+				)
+			);
+		}
+		return $image_placeholder . $video_header;
+	}
+
+	/**
+	 * Replace the header image markup with a header image.
+	 *
+	 * This is JS-driven in Core themes like Twenty Sixteen and Twenty Seventeen.
+	 * So in order for the header video to display,
+	 * this replaces the markup of the header image.
+	 *
+	 * @since 1.0
+	 * @link https://github.com/WordPress/wordpress-develop/blob/d002fde80e5e3a083e5f950313163f566561517f/src/wp-includes/js/wp-custom-header.js#L54
+	 * @param array $atts The image tag attributes.
+	 * @return string $html Filtered markup.
+	 */
+	public static function output_header_image( $atts ) {
+
+		$atts['layout'] = 'fill';
+		unset( $atts['width'] );
+
+		$place_holder = AMP_HTML_Utils::build_tag( 'img', $atts );
```

Comment:
`img` or `amp-img`? I think this should be creating an `amp-img` here.

Reply:
Oops, sorry, it was, I removed it for testing something, but forgot to revert 601c005